### PR TITLE
Sdk template builder

### PR DESCRIPTION
**Pull request by daniel-statsig**
- Repo Links in Title
- Code snippet tabs sync (Select "typescript" changes all tabs to "typescript")
- Right TOC works
- Standardize Client Init Res
- Standardize Local Overrides

### CODE REVIEW

**Comment by jasonwzm** [comment]
🚀 Deployed on https://638fba193832a01028c689f2--cozy-fox-0defec.netlify.app

File: docs/server/golangSDK.mdx [hunk 1]
```diff
@@ -1,8 +1,76 @@
...
+import GetConfig from "./go/_getConfig.mdx";
+import GetExperiment from "./go/_getExperiment.mdx";
+import LogEvent from "./go/_logEvent.mdx";
+import MoreFAQs from "./go/_faqs.mdx";
+import Options from "./go/_options.mdx";
+import OverridesSnippet from "./go/_localOverrides.mdx";
+import ShutdownSnippet from "./go/_shutdown.mdx";
+import * as ReferenceSnippets from "./go/_reference.mdx";
+
+export const Result = SDKDocsBuilder(
```
Comment (daniel-statsig) [inline docs/server/golangSDK.mdx:34]: They all use this Builder pattern now, this ensures that TOC works.

File: docs/client/androidClientSDK.mdx [hunk 2]
```diff
@@ -3,6 +3,65 @@ sidebar_label: Android
...
+import GetConfig from "./Android/_getConfig.mdx";
+import GetExperiment from "./Android/_getExperiment.mdx";
+import LogEvent from "./Android/_logEvent.mdx";
+import UpdateUser from "./Android/_updateUser.mdx";
+import MoreFAQs from "./Android/_faqs.mdx";
+import Options from "./Android/_options.mdx";
+import LocalOverridesSnippet from "./Android/_localOverrides.mdx";
+import ShutdownSnippet from "./Android/_shutdown.mdx";
+
+export const Result = SDKDocsBuilder(
```
Comment (tore-statsig) [inline docs/client/androidClientSDK.mdx:33]: neat

File: docs/client/dotnetSDK.mdx [hunk 3]
```diff
@@ -3,6 +3,67 @@ sidebar_label: .NET
...
+        getConfig: <GetConfig />,
+        getExperiment: <GetExperiment />,
+        logEvent: <LogEvent />,
+      },
+    ],
+    [StatsigUser, { updateUser: <UpdateUser /> }],
+    [StatsigOptions, { options: <Options /> }],
+    [Shutdown, { shutdown: <ShutdownSnippet /> }],
+    // [LocalOverrides, { localOverrides: <LocalOverridesSnippet /> }],
+    // [ManualExposures, { addedInVersion: "1.16.0", ...ManualExposureSnippets }],
```
Comment (tore-statsig) [inline docs/client/dotnetSDK.mdx:63]: nice, I like the addedInVersion prop

File: docs/client/iOS/_checkGate.mdx [hunk 4]
```diff
@@ -20,7 +21,7 @@ if Statsig.checkGate("new_homepage_design") {
   </TabItem>
   <TabItem value="objectivec">
 
-```objectivec
```
Comment (tore-statsig) [inline docs/client/iOS/_checkGate.mdx:23]: I think this is actually the objectivec api - the tab lets you toggle

File: docs/client/iOS/_install.mdx [hunk 5]
```diff
@@ -19,12 +33,18 @@ targets: [
 //...
 ```
 
+  </TabItem>
+  <TabItem value="pods">
+
 If you are using CocoaPods, our pod name is 'Statsig', and you can include the following line to your Podfile:
 
-```
+```ruby
```
Comment (tore-statsig) [inline docs/client/iOS/_install.mdx:41]: the podfile is in ruby?

File: docs/client/iOS/_localOverrides.mdx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+Statsig.removeAllOverrides()
+
+// Returns a StatsigOverrides object, contain all the current overrides
+let currentOverrides = Statsig.getAllOverrides()
+```
+
+  </TabItem>
+  <TabItem value="objectivec">
+
+```swift
```
Comment (tore-statsig) [inline docs/client/iOS/_localOverrides.mdx:38]: objectivec

File: docs/client/iOS/_logEvent.mdx [hunk 7]
```diff
@@ -16,7 +17,7 @@ Statsig.logEvent(withName: "purchase", value: 2.99, metadata: ["item_name": "rem
   </TabItem>
   <TabItem value="objectivec">
 
-```objectivec
+```swift
```
Comment (tore-statsig) [inline docs/client/iOS/_logEvent.mdx:20]: no

File: docs/client/iOS/_manualExposures.mdx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+export const Snippets = {
+  // Gates
+  gateSnippet: (
+    <CodeBlock language="swift">
+      {`// Swift
+let result = Statsig.checkGateWithExposureLoggingDisabled("a_gate_name")
+\t
+// Objective C
+bool result = [Statsig checkGateWithExposureLoggingDisabled:@"a_gate_name"];
+`}
```
Comment (tore-statsig) [inline docs/client/iOS/_manualExposures.mdx:12]: huh, this breaks from the pattern of all the ones above, which actually have a tabbed view for swift and objective c.  You dont need to fix all of this in this PR, but it seems we could follow up with a `Snippets` that supports multiple languages

File: docs/client/iOS/_updateUser.mdx [hunk 9]
```diff
@@ -23,7 +24,7 @@ Statsig.updateUser(newUser) { errorMessage in
   </TabItem>
   <TabItem value="objectivec">
 
-```objectivec
+```swift
```
Comment (tore-statsig) [inline docs/client/iOS/_updateUser.mdx:27]: nah

File: docs/client/js/_checkGate.mdx [hunk 10]
```diff
@@ -1,7 +1,34 @@
...
+  // Gate is on, show new home page
+} else {
+  // Gate is off, show old home page
+}
+```
+
+  </TabItem>
+  <TabItem value="js">
+
 ```jsx
```
Comment (tore-statsig) [inline docs/client/js/_checkGate.mdx:25]: the javascript and typescript api are the same, so why bother?

File: docs/client/js/_faqs.mdx [hunk 11]
```diff
@@ -28,7 +28,7 @@ Taken together, `localMode` and override APIs will help you mock the exact value
 
 We strive to keep the SDK as lightweight as possible, while supporting as many browsers as possible. The primary feature that our SDK relies on which may not be supported by all browsers is a javascript Promise. You may wish to [polyfill](https://developer.mozilla.org/en-US/docs/Glossary/Polyfill) a Promise library to ensure maximum browser compatibility. We recommend [taylorhakes/promise-polyfill](https://github.com/taylorhakes/promise-polyfill) for its small size and compatibility.
 
-The SDK has not been tested on IE.  Note that [Mcrosoft is retiring IE11 in June, 2022](https://blogs.windows.com/windowsexperience/2021/05/19/the-future-of-internet-explorer-on-windows-10-is-in-microsoft-edge/)
+The SDK has not been tested on IE. Note that [Mcrosoft is retiring IE11 in June, 2022](https://blogs.windows.com/windowsexperience/2021/05/19/the-future-of-internet-explorer-on-windows-10-is-in-microsoft-edge/)
```
Comment (tore-statsig) [inline docs/client/js/_faqs.mdx:31]: Microsoft

Comment (tore-statsig) [reply docs/client/js/_faqs.mdx:31]: retired * thats in the past now wow

File: docs/client/js/_localOverrides.mdx [hunk 12]
```diff
@@ -0,0 +1,50 @@
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+import CodeBlock from "@theme/CodeBlock";
+
+<!-- Because its the same for both languages -->
+
+export const Code = ({ type }) => {
+  return (
+    <CodeBlock language="swift">{`// Overrides the given gate to the specified value
```
Comment (tore-statsig) [inline docs/client/js/_localOverrides.mdx:9]: thats not swift.  copypasta

File: docs/client/jsClientSDK.mdx [hunk 13]
```diff
@@ -1,8 +1,75 @@
 ---
 sidebar_label: JavaScript
-title: JavaScript Client SDK for Statsig
```
Comment (tore-statsig) [inline docs/client/jsClientSDK.mdx:3]: lol

File: docs/sdks/_SDKDocsBuilder.mdx [hunk 14]
```diff
@@ -0,0 +1,16 @@
+import Gap from "./_Gap.mdx";
+
+export const SDKDocsBuilder = (sections, toc, appendTOC) => {
+  toc.length = 0;
```
[the remark this answers is not in the document]
Comment (tore-statsig) [reply docs/sdks/_SDKDocsBuilder.mdx:4]: I reviewed so many files just to get here

File: docs/server/Templates/_FAQ.mdx [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+
+#### How do I run experiments for logged out users? {#logged-out-experiments}
+
+See the guide on [device level experiments](/guides/first-device-level-experiment)
+
+#### Is the SDK thread safe (for multi-threaded languages)? {#thread-safety}
+
+Even though we try to make the SDK thread safe whenever we can, we make no guarantees that is always the case.
+Therefore, please make sure to always call Statsig APIs from the same thread.
+
```
Comment (tore-statsig) [inline docs/server/Templates/_FAQ.mdx:11]: nit remove

File: docs/server/Templates/_ManualExposures.mdx [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+    >
+      v{version}+
+    </span>
+  );
+};
+
+## Manual Exposures <Badge version={props.addedInVersion}/>
+
+:::warning
+Manually logging exposures can be tricky and may lead to an imbalance in exposure events. For example, only triggering exposures for users in the Test group of an experiment will imbalance the experiment, making it useless.
```
Comment (tore-statsig) [inline docs/server/Templates/_ManualExposures.mdx:27]: LOLd at "making it useless"

File: docs/server/Templates/_StatsigUser.mdx [hunk 17]
```diff
@@ -0,0 +1,13 @@
+## Statsig User
+
+When calling APIs that require a user, you should pass as much information as possible in order to take advantage of advanced gate and config conditions (like country or OS/browser level checks), and correctly measure impact of your experiments on your metrics/events. The `userID` field is required because it's needed to provide a consistent experience for a given user (click [here](/server/concepts/user#why-is-statsiguser-with-a-userid-required-for-server-sdks) to understand further why it's important to always provide a userID).
```
Comment (tore-statsig) [inline docs/server/Templates/_StatsigUser.mdx:3]: userid is not required - userid or a custom id is required

File: docs/server/Templates/_StatsigUser.mdx [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+
+Besides `userID`, we also have `email`, `ip`, `userAgent`, `country`, `locale` and `appVersion` as top-level fields on StatsigUser. In addition, you can pass any key-value pairs in an object/dictionary to the `custom` field and be able to create targeting based on them.
+
+<div>{props.statsigUser ?? ""}</div>
+
+### Private Attributes
+
+Have sensitive user PII data that should not be logged? No problem, we have a solution for it! On the StatsigUser object we also have a field called `privateAttributes`, which is a simple object/dictionary that you can use to set private user attributes. Any attribute set in `privateAttributes` will only be used for evaluation/targeting, and removed from any logs before they are sent to Statsig server.
```
Comment (tore-statsig) [inline docs/server/Templates/_StatsigUser.mdx:11]: again, remove the "we have a solution for it!"

File: docs/server/Templates/_WorkingWith.mdx [hunk 19]
```diff
@@ -0,0 +1,31 @@
+import AsyncDisclaimer from "./_AsyncDisclaimer.mdx";
+
+## Working with the SDK
+
+### Checking a Gate
```
Comment (tore-statsig) [inline docs/server/Templates/_WorkingWith.mdx:5]: hmm this is interesting,  It sort of combines the guide with the sdk docs. Not totally sold on that but this is great for now

File: docs/server/java/_faqs.mdx [hunk 20]
```diff
@@ -1,16 +1,11 @@
 #### How can I mock Statsig for testing?
 
-The python server SDK, starting in version 1.18.0+, supports a few features to make testing easier.
+First, there is a `StatsigOption` parameter called `localMode`.
```
Comment (tore-statsig) [inline docs/server/java/_faqs.mdx:3]: StatsigOptions

File: docs/server/java/_install.mdx [hunk 21]
```diff
@@ -1,23 +1,21 @@
...
+```swift
 repositories {
     mavenCentral()
 }
 ```
 
 Then add the dependency:
 
-```
+```swift
```
Comment (tore-statsig) [inline docs/server/java/_install.mdx:13]: wow you really just think that everything is swift, dont you

File: docs/server/java/_localOverrides.mdx [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+  groupId="java-server-sdk-lang-choice"
+  defaultValue="java"
+  values={[
+    { label: "Java", value: "java" },
+    { label: "Kotlin", value: "kotlin" },
+  ]}
+>
+<TabItem value="java" >
+
+```swift
```
Comment (tore-statsig) [inline docs/server/java/_localOverrides.mdx:15]: swift kick in the

File: docs/server/python/_logEvent.mdx [hunk 23]
```diff
@@ -1,4 +1,4 @@
-```python
+```py
```
Comment (tore-statsig) [inline docs/server/python/_logEvent.mdx:1]: sure its not swift?

**Pull request merged**